### PR TITLE
JavaScript Lizmap Config Parser

### PR DESCRIPTION
**Pull request by rldhont**
To wrap the Lizmap config with classes and simplify the JavaScript codes of Lizmap. 

Funded by 3Liz

To get access to the config, it is available through `lizMap.mainLizmap.initialConfig` so you can do:
```js
// get options config instance
lizMap.mainLizmap.initialConfig.options

// get layers config instance
lizMap.mainLizmap.initialConfig.layers
// The layer names
lizMap.mainLizmap.initialConfig.layers.layerNames
// The layer ids
lizMap.mainLizmap.initialConfig.layers.layerIds
// iterate through base layers config
for (const layerCfg of lizMap.mainLizmap.initialConfig.layers.getLayerConfigs()) {
    // The layer type: layer, group
    layerCfg.type
    // The layer name
    layerCfg.name
    // The layer id: it is the same has name for group instead it is the qgis layer id
    layerCfg.id
    // The base layer title
    layerCfg.title
}
// Get layer config by layer name
lizMap.mainLizmap.initialConfig.layers.getLayerConfigByLayerName('null_island OGRGeoJSON Point')
// Get layer config by layer id
lizMap.mainLizmap.initialConfig.layers.getLayerConfigByLayerId('null_island20200414115730489'))

// get base layers config instance
lizMap.mainLizmap.initialConfig.baseLayers
// The stratup base layer name
lizMap.mainLizmap.initialConfig.baseLayers.startupBaseLayerName
// The base layer names
lizMap.mainLizmap.initialConfig.baseLayers.baseLayerNames
// iterate through base layers config
for (const baseLayerCfg of lizMap.mainLizmap.initialConfig.baseLayers.getBaseLayerConfigs()) {
    // The base layer type: empty, xyz, bing, wmts, lizmap
    baseLayerCfg.type
    // The base layer name
    baseLayerCfg.name
    // The base layer title
    baseLayerCfg.title
}

// and the other config
```

### CODE REVIEW

File: assets/src/modules/config/Theme.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
+import { BaseObjectConfig } from './Base.js';
+import { ValidationError } from './../Errors.js';
+
+const layerThemeRequiredProperties = {
+    'style': {type: 'sting'},
```
Comment (nboisteault) [inline assets/src/modules/config/Theme.js:5]: `string`

Comment (nboisteault) [reply assets/src/modules/config/Theme.js:5]: TypeScript is used?

Comment (rldhont) [reply assets/src/modules/config/Theme.js:5]: @nboisteault no typescript only javascript.

File: assets/src/modules/config/Metadata.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { BaseObjectConfig } from './Base.js';
+
+const optionalProperties = {
+    'lizmap_plugin_version_str': {type: 'string', default: '3.1.8'},
```
Comment (Gustry) [inline assets/src/modules/config/Metadata.js:4]:
```suggestion
    // Lizmap plugin 3.1.8 is the minimum version when these keys were included in the CFG file
    // At that time, it was LWC 3.2 the stable version.
    'lizmap_plugin_version_str': {type: 'string', default: '3.1.8'},
```

**Review by Gustry — COMMENTED** [review]
Do you know which tickets this PR might fix/be related to ? There might be a few maybe.
In the middle term, it will close #925 I guess ?

File: assets/src/modules/config/Metadata.js (hunk 2 above)
Comment (rldhont) [reply assets/src/modules/config/Metadata.js:6]: Fixes

**Comment by rldhont** [comment]
All tests passed!

**Pull request merged**